filter_plugins/filter_hosts.py

```diff
@@ -78,6 +78,14 @@ def get_instance_systemd_service(app_name, inventory_hostname, stateboard):
     return '{}@{}'.format(app_name, inventory_hostname)
 
 
+def is_stateboard_on_machine(hosts, hostvars, inventory_hostname):
```

Comment:
I'm not sure that this check is needed. This role interprets each instance as Ansible host, so you don't need to run stateboard-specific tasks for each instance(Ansible host) placed on the same machine(physical machine, not Ansible host) as statebord. All you need is `stateboard` flag that indicated that tasks are running for stateboard instance(Ansible host).

Reply:
`install_targz.yml` is not running on a stateboard instance (according to `get_one_not_expelled_instance_on_machine` filter). This check is needed to know when are we dealing with host, that is on the same machine as stateboard.

I do not know though, what to do when stateboard is on a separate machine... but that scenario has never been supported, so we can open an issue, I guess

Reply:
You don't need to run `Configure stateboard` tasks. `stateboard_work_dir` and `stateboard_work_dir` are created on systemd service start (see unit file). I suggest you to place stateboard unit file always when TGZ is deployed (see comment below).

BTW, `get_one_not_expelled_instance_for_machine` works for stateboard on separate machine. I've just deployed this configuration  (see `molecule/default/hosts.yml`, I've removed all other instances and replicasets configurations):

```yaml
...
        host1:
          vars:
            ansible_host: vm1
            cartridge_package_path: ./packages/myapp-1.0.0-0.rpm

          hosts:
            my-stateboard:

        host2:
          vars:
            ansible_host: vm2
            cartridge_package_path: ./packages/myapp-1.0.0-0.deb

          hosts:
            core-1:
```